### PR TITLE
Fix unknown keyword argument passed to accuracy in train.evaluate.

### PR DESCRIPTION
**Pull request by muammar**
When performing multiclass training, there is a problem with the call of
`accuracy` metric where the `labels` argument is incorrectly passed to
the function, throwing an error.

### CODE REVIEW

**Comment by hesther** [comment]
Dear @muammar ,
Thank you very much for contributing to Chemprop! You are right, evaluating a multi class model via accuracy currently throws an error. However, your fix is not quite right yet. By simply omitting the `label=`, the error for evaluating the accuracy disappears... however, this causes the metric cross_entropy (see [here](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.log_loss.html)) to be evaluated wrongly, since the labels are not the third variable in the function call (which you imply by omitting `label=`). Instead, you would need to introduce an additional if/else statement or modify the existing one, e.g. via `if dataset_type == 'multiclass' and metric == 'cross_entropy'`. This would be ok since only cross_entropy actually requires the labels to be set (accuracy does not, and all other implemented metrics are not compatible with multi class models anyways). There are also a number of other options on how to achieve the same behavior, but I guess that would be the easiest.

If you have time to make these changes, you can squash your two commits (the existing one, and the next one implementing the changes) into one, and I can merge them. If not, I can also close this pull request and open a new one!

**Comment by muammar** [comment]
@hesther thanks for your reply. Sure, I will make those changes and update the PR. I will ping you back once it is done.

**Comment by muammar** [comment]
@hesther I pushed f0de2fbd71184cecb82a1a83051adc920f6ae3c0, and hopefully, that covers correctly what you pointed out in your message.

**Comment by hesther** [comment]
@muammar Just had a look, I think this is still not quite correct (which is my fault, I should have been more accurate with my first answer. Namely, I wrote that by omitting `labels=` the 'error goes away' which was a rather silly/wrong way to put it. The error message disappears but what happens under the hood is wrong because the third function argument in the accuracy call is supposed to be a float, namely the threshold above which a prediction is 1, otherwise 0 (and not a list of labels)... but again, I should have put this more clear in my first answer, I apologize!). What you need is something like

```
if dataset_type == 'multiclass' and metric == 'cross_entropy':
    results[metric].append(metric_func(valid_targets[i], valid_preds[i],
                                                    labels=list(range(len(valid_preds[i][0])))))
else:
    results[metric].append(metric_func(valid_targets[i], valid_preds[i])
```
because that provides the labels argument to the sklearn log_loss function, which is really the only function that needs the labels (other metrics only take the targets and preds as input, see chemprop/utils.py)

**Comment by muammar** [comment]
@hesther no worries! Is 4190edb1c7737cea596bb09c34143127f9a5b687 good now?

**Comment by hesther** [comment]
Looks great, thanks! Could you squash the three commits into one?

If you are not familiar with doing so:
`git rebase -i HEAD~3`
this will open an editor and show the three latest commits (starting with the newest one). Put the second and third on "s" instead of "pick". Save and close. This will open up a new editor which shows the three latest commit messages. Since the changes were all super small and bug fixes I suggest you comment them out apart from the first commit message (Fix unknown keyword argument passed to accuracy in train.evaluate). Save and close. Use
`git log`
to check whether the commit history is what you want it to be (a single commit instead of three individual commits). If it is ok, push your changes. Since your repo has now diverged from the upstream repo, you need to force push:
`git push -f`

**Comment by hesther** [comment]
I would normally do this for you, if it was a branch on the main chemprop site, but since it isn't I cannot push to your repo. Let me know if you run into any troubles and I can show you different solutions, too.

**Comment by muammar** [comment]
Now it's done with a837a5a803c90a8d15651889ab0a7a7992718a61.

**Comment by hesther** [comment]
Great, I will approve running GitHub workflows now (which runs some integration tests), and as soon as the tests have past (which they of course will, I just need to wait for them until I can merge the branch), I will merge this! Thanks again for contributing!

**Review by hesther — APPROVED** [review]

**Comment by hesther** [comment]
@muammar Just merged! Feel free to delete the branch from now on

**Pull request merged**
